server/controllers/finance/distributionFeeCenter/automatic.js

```diff
@@ -0,0 +1,74 @@
+/**
+* Distribution Fee Center Automatic Controller
+*
+* This function allows automatic distribution of invoices whose services are linked to Principal fee centers
+*/
+const db = require('../../../lib/db');
+const NotFound = require('../../../lib/errors/NotFound');
+
+function automatic(req, res, next) {
+  const { data } = req.body;
+  const transUuids = data.map(item => db.bid(item.uuid));
+
+  const sql = `
+    SELECT BUID(gl.uuid) AS row_uuid, gl.trans_id, gl.debit_equiv, gl.credit_equiv,
+    gl.account_id, gl.record_uuid, sfc.fee_center_id, iv.description, iv.service_id, s.name AS serviceName
+    FROM general_ledger AS gl
+    JOIN invoice AS iv ON iv.uuid = gl.record_uuid
+    JOIN service AS s ON s.id = iv.service_id
+    JOIN service_fee_center AS sfc ON sfc.service_id = s.id
+    WHERE gl.uuid IN (?)
+  `;
+
+  db.exec(sql, [transUuids])
+    .then((rows) => {
+      const dataToDistribute = [];
+      const userId = req.session.user.id;
+
+      rows.forEach((row) => {
+        data.forEach((item) => {
+          if (row.row_uuid === item.uuid) {
+            dataToDistribute.push([
```

Comment:
Is it possible to just use the `rows` array like : 
```js
rows.forEach(row => {
  dataToDistribute.push([
    ...
  ]);
});
```
Because it seems like the `rows` array contains already data defined by this condition : 
```js
if (row.row_uuid === item.uuid)
```

Reply:
After analyzing the codes, the first element ```` rows ````

The second one looks for invoices related to trans Uuid, and for the automatic distribution there is place of combined some elements of QBC and CDF